third-party/realdds/src/dds-notification-server.cpp

```diff
@@ -0,0 +1,138 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-notification-server.h>
+
+#include <realdds/dds-participant.h>
+#include <realdds/dds-publisher.h>
+#include <realdds/dds-stream-server.h>
+#include <realdds/dds-utilities.h>
+#include <realdds/topics/dds-topics.h>
+#include <realdds/dds-topic.h>
+#include <realdds/dds-topic-writer.h>
+
+#include <fastdds/dds/topic/Topic.hpp>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <librealsense2/utilities/concurrency/concurrency.h>
+
+
+using namespace eprosima::fastdds::dds;
+
+namespace realdds {
+
+
+dds_notification_server::dds_notification_server( std::shared_ptr< dds_publisher > const & publisher, const std::string & topic_name )
+    : _publisher( publisher )
+    , _notifications_loop( [this]( dispatcher::cancellable_timer timer ) {
+        if( _active )
+        {
+            // We can send 2 types of notifications:
+            // * Initialize notifications that should be sent for every new reader (like sensors & profiles data)
+            // * Instant notifications that can be sent upon an event and is not required to be
+            //   resent to new readers
+            std::unique_lock< std::mutex > lock( _notification_send_mutex );
+            _send_notification_cv.wait( lock, [this]() {
+                return ! _active || _send_init_msgs || _new_instant_notification;
+            } );
+
+            if( _active && _send_init_msgs )
+            {
+                send_init_msgs();
+                _send_init_msgs = false;
+            }
+
+            if( _active && _new_instant_notification )
+            {
+                // Send all instant notifications
+                topics::raw::device::notification notification;
+                while( _instant_notifications.dequeue( &notification, 1000 ) )
```

Comment:
Use named constants like `one_second_timeout` instead of "magic numbers"

Reply:
Not new code.
But I changed to:
```
                constexpr unsigned timeout_in_ms = 1000;
                while( _instant_notifications.dequeue( &notification, timeout_in_ms ) )
```

Reply:
(will be in next PR)